### DISCUSSION TITLE
Resume experiment if browser/window closed?

### DISCUSSION THREAD

**Discussion by bjmeagher** [jspsych/jsPsych#914] [Q&A]
If possible, I'd like to set up my experiment such that the participant could pick up where they left off if their session is interrupted. I figure if a participant loses their internet service, accidentally closes out of the window, etc., they should have the option to resume their progress, rather than having to start over from scratch.

Is there a way of doing this? My initial thought was to record the number of trials the participant had completed to my database, then put them back into the appropriate spot in their timeline based on that number. However, it looks like the timeline itself doesn't contain enough information to make this a viable solution, especially since I'm making heavy use of randomization and repetition.

I'm pretty new to the world of web-based experimenting, so any advice would be appreciated!

**Comment by vijaymarupudi** [comment]
There are ways to do this, but they are all relatively complicated. One way is to save the data serverside (after each trial). Another way is to save the data locally (using [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)). Then, you can call `jsPsych.init()`, and then call `jsPsych.finishTrial(...)` with each trial's data that you retrieved from storage. That'll leave the participants where they left off.
